src/extension/background-script/connectors/alby.ts

```diff
@@ -0,0 +1,352 @@
+    };
+  }
+
+  signMessage(args: SignMessageArgs): Promise<SignMessageResponse> {
+    // signMessage requires proof of ownership of a non-custodial node
+    // this is not the case in the Alby connector which connects to Lndhub
+    throw new Error(
+      "SignMessage is not supported by Alby accounts. Generate a secret key to use LNURL auth."
+    );
+  }
+
+  async makeInvoice(args: MakeInvoiceArgs): Promise<MakeInvoiceResponse> {
+    const data = await this._request((client) =>
+      client.createInvoice({
+        amount: parseInt(args.amount.toString()),
+        description: args.memo,
+      })
+    );
+
+    return {
+      data: {
+        paymentRequest: data.payment_request,
+        rHash: data.payment_hash,
+      },
+    };
+  }
+
+  private async authorize(): Promise<auth.OAuth2User> {
+    try {
+      const clientId = process.env.ALBY_OAUTH_CLIENT_ID;
+      const clientSecret = process.env.ALBY_OAUTH_CLIENT_SECRET;
+      if (!clientId || !clientSecret) {
+        throw new Error("OAuth client credentials missing");
+      }
+
+      const redirectURL = browser.identity.getRedirectURL();
+      const authClient = new auth.OAuth2User({
+        request_options: this._getRequestOptions(),
+        client_id: clientId,
+        client_secret: clientSecret,
+        callback: redirectURL,
+        scopes: [
+          "invoices:read",
+          "account:read",
+          "balance:read",
+          "invoices:create",
+          "invoices:read",
+          "payments:send",
+          "transactions:read", // for outgoing invoice
+        ],
+        token: this.config.oAuthToken, // initialize with existing token
+      });
+
+      if (this.config.oAuthToken) {
+        try {
+          if (authClient.isAccessTokenExpired()) {
+            const token = await authClient.refreshAccessToken();
+            await this._updateOAuthToken(token.token);
+          }
+          return authClient;
+        } catch (error) {
+          // if auth token refresh fails, the refresh token has probably expired or is invalid
+          // the user will be asked to re-login
+          console.error("Failed to request new auth token", error);
+        }
+      }
+
+      let authUrl = authClient.generateAuthURL({
+        code_challenge_method: "S256",
+      });
+      // TODO: make authorize URL in alby-js-sdk customizable
+      if (process.env.ALBY_OAUTH_AUTHORIZE_URL) {
+        authUrl = authUrl.replace(
+          "https://getalby.com/oauth",
+          process.env.ALBY_OAUTH_AUTHORIZE_URL
+        );
+      }
+      authUrl += "&webln=false"; // stop getalby.com login modal launching lnurl auth
+      const authResult = await this.launchWebAuthFlow(authUrl);
+      const code = new URL(authResult).searchParams.get("code");
+      if (!code) {
+        throw new Error("Authentication failed: missing authResult");
+      }
+
+      const token = await authClient.requestAccessToken(code);
+      await this._updateOAuthToken(token.token);
```

Comment:
We need to make sure the oAuth token is for the same account ID as it was originally created, otherwise show an error message and ask them to try again

"You logged into the wrong account. Please try again"

Reply:
decided not do do this, instead updating the account to match the data from the api